src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py

```diff
@@ -236,17 +238,32 @@ def get_backup(job_parameters: JobParameters) -> Optional[SimulationContext]:
 
         # Use the query method to find rows that match the lookup parameters
         run_ids = pickle_metadata.query(query_conditions)["job_id"].to_list()
-        if run_ids:
-            filename = Path(backup_dir / run_ids[0]).with_suffix(".pkl")
-            with open(filename, "rb") as f:
+        possible_pickles = [
+            Path(backup_dir / run_id).with_suffix(".pkl") for run_id in run_ids
+        ]
+        existing_pickles = [pickle for pickle in possible_pickles if pickle.exists()]
+        if existing_pickles:
+            last_pickle = max(existing_pickles, key=os.path.getctime)
+            if len(existing_pickles) > 1:
+                logger.warning(
+                    f"Multiple backups found for {job_parameters}. Using the most recent and deleting the rest."
+                )
+                for stale_file in set(existing_pickles) - {last_pickle}:
+                    os.remove(stale_file)
+            with open(last_pickle, "rb") as f:
                 sim = dill.load(f)
+            current_job_id = get_current_job().id
+            logger.info(f"Renaming backup file {last_pickle} to {current_job_id}.pkl")
+            if job_parameters.backup_configuration["backup_freq"] is not None:
```

Comment:
When would this ever not be the case? I guess if you ran with backups, and then restarted without them? Is that a realistic use-case?

Reply:
>  if you ran with backups, and then restarted without them
Yes. In particular, if we don't use backups, then we don't make / append to the metadata file for that set of jobs. So if we rename the file, a subsequent run would not find the backup. 

Whether or not it's realistic, it's possible, so we should handle it consistently. The other option, though, is to just always write/append jobs to the backup csv file regardless of whether we're actually planning to save backups or not.